### ISSUE TITLE
API to get selected content in a cell

### ISSUE THREAD

**Issue by sharky93** [facebook/fixed-data-table#186]
As per the docs, I don't see an easy way to fetch the text that has been selected in a cell in the table.

**Comment by ehzhang** [comment]
Hi @sharky93, what do you mean by fetching text that has been selected in a cell?

**Comment by sharky93** [comment]
Yep, so, `window.getSelection()` is a way, I was wondering if that is the best way?

**Comment by ehzhang** [comment]
Can you explain what you're trying to accomplish? If you are trying to get data from a highlighted selection, `window.getSelection()` is a perfectly valid method.

**Comment by sharky93** [comment]
I'm trying to get cell data on the mouseUp event. There is no direct method for cells, but onRowClick works just fine.

**Comment by sharky93** [comment]
Also, do we have a convenient way to get the rowCount number or some detail about the column for the row clicked on?

**Comment by ehzhang** [comment]
rowCount should be consistent for the entire table - one column should not have more or less. From the arguments in a  rowClickHandler, you won't be able to access them, but considering you have to provide a rowsCount to the table itself, you should be able to access it from there.

**Comment by sharky93** [comment]
@ehzhang let me rephrase, my bad. I'm looking for a way to get the row index of the cell which has been clicked on and similarly, the column index.

**Comment by ehzhang** [comment]
from the onRowClick prop, row index is the second argument. This does not also provide the column index, unfortunately.

**Comment by sharky93** [comment]
does it sound like something that could be added to the API in recent future?

**Comment by Lashas83** [comment]
I actually have made API change to include onCellClick and onCellDoubleClick methods in this pull request: https://github.com/facebook/fixed-data-table/pull/188.

**Comment by ehzhang** [comment]
This will very likely not be something that will be added to the API in the future. Instead, we are looking to cut down on the API so you can more flexibly attach these onClick and onDoubleClick listeners (or anything) without depending on the Column or Table API.

**Comment by ehzhang** [comment]
I'm going to be closing this issue as providing an API for this information is not in the scope of the table's responsibility.
